website/docs/cloud-docs/workspaces/variables/index.mdx

```diff
@@ -60,7 +60,22 @@ module "ec2_instances" {
 }
 ```
 
-If a required input variable is missing, Terraform plans in the workspace will fail and print an explanation in the log.
+If a required input variable is missing, Terraform plans in the workspace fail and print an explanation in the log. 
+
+#### Ephemeral variables
+
+-> **Note**: Ephemeral variables are available in Terraform v1.10.0 and later. [Learn more about ephemeral variables](/terraform/language/values/variables#exclude-values-from-state).
+
+Defining a variable as `ephemeral` in your configuration makes it available during runtime, but Terraform omits ephemeral values from state and plan files.
+
+```hcl
+variable "session_token" {
+  type      = string
+  ephemeral = true
+}
+```
+
+If an `ephemeral` variable in your configuration references a Terraform variable, you must mark your workspace's Terraform variable as [sensitive](/terraform/cloud-docs/workspaces/variables/managing-variables#sensitive-values) to hide its value in the HCP Terraform UI.
```

Comment:
I don't entirely understand what this sentence is trying to say. The part I don't understand is how "ephemeral variable ... references a Terraform variable" - variables do not reference each other unless inside of validation blocks - but then I don't understand the point about sensitivity.

Reply:
I'll be honest, I struggled with how to frame this because I'm finding it hard to understand why you'd need to create Terraform variables in HCP TF for ephemeral values. @radeksimko do you have any suggestions on how to present this to folks? Or @robin-norwood?

Reply:
Oh, I think I _now_ understand what you were trying to say. 😅 

The reality is that currently variables get implicitly persisted somewhere in HCP, regardless of whether they are ephemeral or not. This is however separate from the state or plan file.

I don't know enough about HCP's permission/ACL model to know what _exact_ difference does it make but I believe that variables are treated differently on the backend from state or plan files, so it's not strictly speaking seen as a violation of the promise.

All that is to say I can't think of specific use cases though right now - which is also why we avoided any special treatment in HCP TF.

Reply:
Thanks for the context! Do you think we need to call out `ephemeral` variables in HCP TF at all then? Or do you feel like we don't need this section? Tagging in @brandonc in case you have opinions from the HCP TF side of things!

Reply:
I would just avoid mentioning it for now.